.github/workflows/release.yml

```diff
@@ -235,22 +257,21 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-pip-
       - uses: actions/setup-dotnet@v1
-        if: ${{ matrix.language == 'dotnet' }}
+        if: ${{ needs.get-smoke-test-configuration.outputs.language == 'dotnet' }}
         with:
           dotnet-version: '3.1.x'
       - name: download layer tf file
         uses: actions/download-artifact@v2
         with:
           name: layer.tf
       - name: overwrite layer.tf
-        if: ${{ github.event.inputs.layer_kind == matrix.layer_kind }}
```

Comment:
@wangzlei my solution to this problem was actually just going to be to copy this condition to every step of the smoke test workflow :)

Any idea why it was originally only added to this step and not any others? Don't want to miss anything on this

Reply:
Originally the smoke test ran every layer kind, but only one of them gets updated in release workflow. So, only override the released layer kind, the others were useless as this issue said.
Now this PR runs smoke test for the released layer kind only, we can safely remove this condition.

Reply:
Yeah this is the critical step: Publish the layer. We only wanted to do actually publish if this Layer was the one that we were releasing once the smoke test passed.

Reply:
Well the layer is published as part of the above `publish-prod` job right? This step just made sure we actually run the test against the newly published layer.

Related thought, since we're automatically overwriting the `layer.tf` file here, we could just open a PR as part of the workflow too and make it one less manual step for us. Only problem again is if there's multiple sample apps, only 1 would get updated.